### ISSUE TITLE
🧪 Property-Based Testing: Numerical stability and invariants

### ISSUE THREAD

**Issue by mikkihugo** [mikkihugo/claude-code-zen#233]
## 📋 Summary

Implement property-based testing using proptest to ensure numerical stability, validate invariants, and test edge cases across the neural network codebase.

### Areas for Property Testing:
- [ ] **Numerical Stability**
  - Gradient calculations
  - Activation functions
  - Loss functions
  - Matrix operations

- [ ] **Model Invariants**
  - Output dimensions
  - Parameter bounds
  - State consistency

- [ ] **Data Processing**
  - Scaling/normalization reversibility
  - Window generation correctness
  - Missing value handling

### 1. Numerical Properties
```rust
use proptest::prelude::*;

proptest! {
    #[test]
    fn test_activation_bounded(x in any::<f64>()) {
        let sigmoid = sigmoid_activation(x);
        prop_assert!(sigmoid >= 0.0 && sigmoid <= 1.0);
    }
}
```

### 2. Model Properties
- Forward pass determinism
- Gradient non-explosion
- Parameter update stability
- Loss monotonicity

### 3. Data Invariants
- TimeSeriesDataset consistency
- Preprocessing reversibility
- Feature engineering correctness

### 4. Fuzz Testing
- Random input generation
- Malformed data handling
- Extreme value testing

### 5. Statistical Properties
- Distribution preservation
- Correlation maintenance
- Forecast bounds

## 🛠️ Technical Approach

```toml
[dev-dependencies]
proptest = "1.6"
arbitrary = { version = "1.4", features = ["derive"] }
proptest-derive = "0.5"
```

### Custom Generators
- Time series data generators
- Neural network input generators
- Configuration generators

## 📊 Success Criteria
- [ ] All numerical operations tested for stability
- [ ] Model invariants validated
- [ ] Edge cases discovered and handled
- [ ] No panics on random inputs

## 🔗 Parent Issue
Part of #229
